validate.go

```diff
@@ -393,7 +393,7 @@ func checkMandatoryUnit(field reflect.Value, tagField reflect.StructField, paren
 			return msgs, false
 		}
 	case reflect.Slice:
-		if mandatory && (field.Len() == 0) {
+		if mandatory && (field.IsNil() || (field.Len() == 0)) {
```

Comment:
You [don't need those parens][1] around the `Len == 0` bit.

The `IsNil` addition looks ok, but the docs say that it differs from `== nil` by [being more panicky][2], so I think I'd prefer `== nil`.

[1]: https://golang.org/ref/spec#Operator_precedence
[2]: https://golang.org/pkg/reflect/#Value.IsNil